cmd/kubeapps-apis/server/packages.go

```diff
@@ -118,6 +123,139 @@ func (s packagesServer) GetAvailablePackageDetail(ctx context.Context, request *
 	}, nil
 }
 
+// GetInstalledPackageSummaries returns the installed package summaries based on the request.
+func (s packagesServer) GetInstalledPackageSummaries(ctx context.Context, request *packages.GetInstalledPackageSummariesRequest) (*packages.GetInstalledPackageSummariesResponse, error) {
+	contextMsg := ""
+	if request.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.Context.Cluster, request.Context.Namespace)
+	}
+
+	log.Infof("+core GetInstalledPackageSummaries %s", contextMsg)
+
+	// Aggregate the response for each plugin
+	pkgs := []*packages.InstalledPackageSummary{}
+	// TODO: We can do these in parallel in separate go routines.
+	for _, p := range s.plugins {
+		response, err := p.server.GetInstalledPackageSummaries(ctx, request)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Invalid GetInstalledPackageSummaries response from the plugin %v: %v", p.plugin.Name, err)
+		}
+
+		// Add the plugin for the pkgs
+		pluginPkgs := response.InstalledPackageSummaries
+		for _, r := range pluginPkgs {
+			if r.InstalledPackageRef == nil {
+				r.InstalledPackageRef = &packages.InstalledPackageReference{}
+			}
+			r.InstalledPackageRef.Plugin = p.plugin
+		}
+		pkgs = append(pkgs, pluginPkgs...)
+	}
+
+	// Build the response
+	// TODO: Sort via default sort order or that specified in request.
```

Comment:
// TODO: pagination support (we still need to update the existing pagination in helm to use item offsets rather than page offsets to do this)

Reply:
Indeed. In fact, we need `GetAvailablePackagesSummaries` to be paginated; otherwise, the catalog just won't work (I mean, well fetch the first `n` elements).
I've implemented a dirty initial way just fetching all the plugin's responses w/o pagination, and then, paginate the aggregated slice.